### PR TITLE
Trying new approach

### PR DESCRIPTION
**Pull request by Daboo1998**

### CODE REVIEW

**Review by PAKovalev — CHANGES_REQUESTED** [review]

File: packages/sdk/src/interfaces/plugin.interface.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {
+  SpacePluginPropsInterface,
+  WidgetPluginPropsInterface,
+  HomePluginPropsInterface
+} from './pluginProps.interface';
+
+export interface PluginInterface {
+  SpaceApp?: React.FC<SpacePluginPropsInterface>;
```
[the remark this answers is not in the document]
Comment (Daboo1998) [reply packages/sdk/src/interfaces/plugin.interface.ts:8]: Well we need to differentiate somehow, plugin is the whole thing while each part (Space, Widget and Home) is an app. We need to mayve change PluginType to AppType not to confuse

File: packages/app/src/shared/hooks/pluginLoader.tsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import {ErrorBoundary} from '@momentum/ui-kit';
 import {useDynamicScript} from './useDynamicScript';
 
 const loadComponent =
-  (scope: string, module: string) =>
+  (scope: string, pluginType: PluginTypeEnum) =>
```
Comment (dmitry-yudakov) [inline packages/app/src/shared/hooks/pluginLoader.tsx:9]: It's ok this way for the moment but in future I'd leave this function to return the whole plugin object and inside this object we would look for Space, Widget, etc components but possibly also for Space Tab Icon component or Widget Button component or render function.

File: packages/sdk/src/interfaces/plugin.interface.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmitry-yudakov) [reply packages/sdk/src/interfaces/plugin.interface.ts:8]: Could be SpaceTabContent or WidgetComponent. I don't think any Type is needed at all.

**Review by dmitry-yudakov — APPROVED** [review]
I like it, I think it's a step in the right direction.

File: packages/sdk/src/interfaces/plugin.interface.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Daboo1998) [reply packages/sdk/src/interfaces/plugin.interface.ts:8]: No we have a typo for the FE to use when loading. Not talking about the component itself

[the remark this answers is not in the document]
Comment (dmitry-yudakov) [reply packages/sdk/src/interfaces/plugin.interface.ts:8]: Well, it's the implementation detail of the host app. The interface which will be used by 3rd party devs should be clear for them.

[the remark this answers is not in the document]
Comment (Daboo1998) [reply packages/sdk/src/interfaces/plugin.interface.ts:8]: we need to defrentiate the whole thing from each part, so we need to think how we want to name each. We cannot call the whole app and its parts a plugin

**Pull request merged**